subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/spock/Spock2IntegrationSpec.groovy

```diff
@@ -30,6 +30,12 @@ class Spock2IntegrationSpec extends AbstractIntegrationSpec {
             ${mavenCentralRepository()}
             dependencies {
                 testImplementation("org.spockframework:spock-core:$SPOCK_VERSION")
+
+                constraints {
+                    implementation("org.codehaus.groovy:groovy:${GroovySystem.version}") {
+                        because("needs to support the same JDKs as Gradle")
+                    }
+                }
```

Comment:
This isn't quite right. It's not really that we need a version of Groovy that Gradle supports, we need a version of Groovy that's supported by the JDK we're using. 

Could we instead be explicit with the version of Groovy we want here? either make it based on which JDK we're using or just always use a constraint on >3.0.12?

Reply:
I added a text-fixtures dependency on `language-groovy` and re-used the GroovyCoverage logic.